cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -168,7 +192,35 @@ std::unique_ptr<column> generate_clean_offsets(lists_column_view const& lists_co
 }
 
 /**
- * @brief Populate list offsets for all list entries
+ * @brief Transform a given lists column to a new lists column in which all the list entries holding
+ * -NaN value are replaced by (positive) NaN.
+ *
+ * Replacing -NaN by NaN is necessary before sorting (individual) lists because the sorting API is
+ * using radix sort, which compares bits of the number thus it may separate -NaN by NaN to the two
+ * ends of the result column.
+ */
+std::unique_ptr<column> replace_negative_nans_entries(column_view const& lists_entries,
+                                                      lists_column_view const& lists_column,
+                                                      rmm::cuda_stream_view stream)
+{
+  // We need to copy the offsets column of the input lists_column. Since the input lists_column may
+  // be sliced, we need to generate clean offsets (i.e., offsets starting from zero).
+  auto new_offsets =
+    generate_clean_offsets(lists_column, stream, rmm::mr::get_current_device_resource());
```

Comment:
Probably too late for this PR, but having a function that does this for lists (and strings) available as a utility seems like it would be useful.

Reply:
Yes, maybe very useful. I observed that this utility is also implemented in `lists/segmented_sort.cu`. I'll file an issue to add it in the future.

Reply:
Done. Here it is: https://github.com/rapidsai/cudf/issues/9256